spec/unit/server_spec.rb

```diff
@@ -70,6 +70,14 @@ def app
       end
     end
 
+    describe "GET /:id with non-existent id" do
+      it "redirects to landing page if description is not found" do
+        get "/does_not_exist"
+
+        expect(last_response).to be_redirect
+      end
+    end
```

Comment:
It would be good to also have a feature test, which checks, that the error message is actually displayed.